pg_search/src/postgres/customscan/pdbscan/projections/snippet.rs

```diff
@@ -0,0 +1,170 @@
+use crate::api::index::FieldName;
+use crate::index::state::SearchState;
+use crate::nodecast;
+use crate::postgres::customscan::pdbscan::projections::OpaqueRecordArg;
+use pgrx::pg_sys::expression_tree_walker;
+use pgrx::{
+    default, direct_function_call, pg_extern, pg_guard, pg_sys, FromDatum, IntoDatum, PgList,
+};
+use std::ptr::addr_of_mut;
+use tantivy::snippet::SnippetGenerator;
+use tantivy::DocAddress;
+
+#[pg_extern(name = "snippet", stable, parallel_safe)]
+fn snippet_from_relation(
+    _relation_reference: OpaqueRecordArg,
+    field: FieldName,
+    start_tag: default!(String, "'<b>'"),
+    end_tag: default!(String, "'</b>'"),
```

Comment:
We need to port over `max_num_chars` as well: https://docs.paradedb.com/documentation/full-text/highlighting. Also, let's keep the existing names (prefix/postfix) to stay consistent with Tantivy.